packages/phone-number-privacy/signer/test/integration/pnp.test.ts

```diff
@@ -225,33 +264,544 @@ describe('pnp', () => {
+            )
+            .mockRejectedValueOnce(new Error())
+
+          const req = getPnpSignRequest(
+            ACCOUNT_ADDRESS1,
+            BLINDED_PHONE_NUMBER,
+            AuthenticationMethod.WALLET_KEY,
+            IDENTIFIER
+          )
+          const authorization = getPnpRequestAuthorization(req, PRIVATE_KEY1)
+          const res = await sendRequest(req, authorization, SignerEndpoint.PNP_SIGN)
+
+          expect(res.status).toBe(200)
+          expect(res.body).toMatchObject<SignMessageResponseSuccess>({
+            success: true,
+            version: res.body.version,
+            signature: expectedSignature,
+            performedQueryCount: performedQueryCount, // Not incremented
+            totalQuota: expectedQuota,
+            blockNumber: testBlockNumber,
+            warnings: [
+              ErrorMessage.FAILURE_TO_INCREMENT_QUERY_COUNT,
+              ErrorMessage.FAILURE_TO_STORE_REQUEST,
+            ],
+          })
+
+          spy.mockRestore()
+        })
+
+        it('Should return 200 w/ warning on failure to store request', async () => {
+          const spy = jest
+            .spyOn(jest.requireActual('../../src/common/database/wrappers/request'), 'storeRequest')
+            .mockRejectedValueOnce(new Error())
+
+          const req = getPnpSignRequest(
+            ACCOUNT_ADDRESS1,
+            BLINDED_PHONE_NUMBER,
+            AuthenticationMethod.WALLET_KEY,
+            IDENTIFIER
+          )
+          const authorization = getPnpRequestAuthorization(req, PRIVATE_KEY1)
+          const res = await sendRequest(req, authorization, SignerEndpoint.PNP_SIGN)
+
+          expect(res.status).toBe(200)
+          expect(res.body).toMatchObject<SignMessageResponseSuccess>({
+            success: true,
+            version: res.body.version,
+            signature: expectedSignature,
+            performedQueryCount: performedQueryCount + 1,
+            totalQuota: expectedQuota,
+            blockNumber: testBlockNumber,
+            warnings: [ErrorMessage.FAILURE_TO_STORE_REQUEST],
+          })
+
+          spy.mockRestore()
+        })
+
+        it('Should return 500 on bls signing error', async () => {
+          const spy = jest
+            .spyOn(
+              jest.requireActual('../../src/common/bls/bls-cryptography-client'),
+              'computeBlindedSignature'
+            )
+            .mockImplementationOnce(() => {
+              throw new Error()
+            })
+
+          const req = getPnpSignRequest(
+            ACCOUNT_ADDRESS1,
+            BLINDED_PHONE_NUMBER,
+            AuthenticationMethod.WALLET_KEY,
+            IDENTIFIER
+          )
+          const authorization = getPnpRequestAuthorization(req, PRIVATE_KEY1)
+          const res = await sendRequest(req, authorization, SignerEndpoint.PNP_SIGN)
+
+          expect(res.status).toBe(500)
+          // TODO(2.0.0)(Alec): investigate whether we have the intended behavior here
```

Comment:
same as in legacy tests, consider checking the signer DB as well to confirm that the transaction rolled back properly